src/index.spec.ts

```diff
@@ -0,0 +1,41 @@
+import BadgerSDK, { BadgerAPI, BadgerGraph } from '.';
+import { Network } from './config';
+
+const ethAlchemyNodeUrl =
+  'https://eth-mainnet.alchemyapi.io/v2/-8IubgVIbg-UaXks6VLvgsuYujAravb4';
+
+describe('BadgerSDK', () => {
+  let sdk: BadgerSDK;
+  let isSdkReady: boolean | void;
+
+  beforeAll(async () => {
+    // eslint-disable-next-line
+    const initializeFunc = jest.spyOn(BadgerSDK.prototype as any, 'initialize');
+    initializeFunc.mockImplementation(() => new Promise((res) => res(true)));
+
+    sdk = new BadgerSDK({
+      network: Network.Ethereum,
+      provider: ethAlchemyNodeUrl,
+    });
+
+    isSdkReady = Boolean(await sdk.ready());
```

Comment:
might need to think more about this test in particular. we should probably mock the individual service methods ready functions as they call call out to external services for success and if we use this cloudflare public rpc its much for flaky and subject to failures - also we are just relying on a third party for test success

Reply:
ok, no point in this test then

Reply:
i`ll just mock the whole init